src/ring_theory/unique_factorization_domain.lean

```diff
@@ -1333,6 +1354,42 @@ begin
     cases mul_eq_zero.mp h; contradiction }
 end
 
+/-- The only divisors of prime powers are prime powers. -/
+theorem eq_pow_find_of_dvd_irreducible_pow {a p : associates α} (hp : irreducible p)
+  [∀ n : ℕ, decidable (a ∣ p ^ n)] (h : ∃ n : ℕ, a ∣ p ^ n) : a = p ^ nat.find h :=
+begin
+  have := nat.find_spec h,
+  have b_ne_zero := pow_ne_zero _ hp.ne_zero,
+  have a_ne_zero := ne_zero_of_dvd_ne_zero b_ne_zero (nat.find_spec h),
+  haveI : decidable_eq α := by { classical, apply_instance },
```

Comment:
```suggestion
  letI : decidable_eq α := classical.dec_eq _,
```
or similar. Note that `decidable` is data, so usually you want to use `letI` not `haveI` to prevent lean complaining.